awscli/customizations/eks/update_kubeconfig.py

```diff
@@ -235,28 +235,29 @@ def _expand_path(self, path):
 
 
 class EKSClient(object):
-    def __init__(self, session, cluster_name, role_arn, parsed_globals=None):
+    def __init__(self, session, *, parsed_globals=None, parsed_args):
         self._session = session
-        self._cluster_name = cluster_name
-        self._role_arn = role_arn
+        self._cluster_name = parsed_args["cluster_name"]
         self._cluster_description = None
-        self._globals = parsed_globals
+        self._parsed_globals = parsed_globals
+        self._parsed_args = parsed_args
 
-    def _get_cluster_description(self):
+    @property
+    def cluster_description(self):
         """
         Use an eks describe-cluster call to get the cluster description
         Cache the response in self._cluster_description.
         describe-cluster will only be called once.
         """
         if self._cluster_description is None:
-            if self._globals is None:
+            if self._parsed_globals is None:
                 client = self._session.create_client("eks")
             else:
                 client = self._session.create_client(
                     "eks",
-                    region_name=self._globals.region,
-                    endpoint_url=self._globals.endpoint_url,
-                    verify=self._globals.verify_ssl
+                    region_name=self._parsed_globals["region"],
```

Comment:
I believe the `parsed_globals` is still a namespace so we cannot access it as if it has keys; we need to keep it as `self._parsed_globals.region`. This goes for the other places `parsed_globals` are referenced as well.

Reply:
Same observation as above. I was trying to make the use of both variables consistent. But this should be okay if we keep `parsed_args` as `argparse.Namespace`.